src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -323,15 +392,37 @@ public override void Initialize(AnalysisContext context)
                     }
 
                     seenRuleIds.Free();
+                    if (analyzeResourceStrings)
+                    {
+                        RoslynDebug.Assert(localizableTitles != null);
+                        RoslynDebug.Assert(localizableMessages != null);
+                        RoslynDebug.Assert(localizableDescriptions != null);
+                        RoslynDebug.Assert(resourcesDataValueMap != null);
+
+                        FreeLocalizableStringsMap(localizableTitles);
+                        FreeLocalizableStringsMap(localizableMessages);
+                        FreeLocalizableStringsMap(localizableDescriptions);
+                        resourcesDataValueMap.Free();
+                    }
                 });
             });
+
+            static void FreeLocalizableStringsMap(PooledLocalizabeStringsConcurrentDictionary localizableStrings)
+            {
+                foreach (var builder in localizableStrings.Values)
+                {
+                    builder.Free();
+                }
+
+                localizableStrings.Free();
+            }
         }
 
         private static bool TryGetDescriptorCreateMethodAndArguments(
             IFieldInitializerOperation fieldInitializer,
             INamedTypeSymbol diagnosticDescriptorType,
-            out IMethodSymbol creationMethod,
-            out ImmutableArray<IArgumentOperation> creationArguments)
+            [NotNullWhen(returnValue: true)] out IMethodSymbol? creationMethod,
+            [NotNullWhen(returnValue: true)] out ImmutableArray<IArgumentOperation> creationArguments)
```

Comment:
Not sure to understand what `NotNullWhen(true)` brings on this parameter.

Reply:
I thought it ensures the array is not default?